### ISSUE TITLE
[Enhancement]: Modules / Compose - ability to Attach to services when they are started via Up.

### ISSUE THREAD

**Issue by nehemming** [testcontainers/testcontainers-go#2567]
### Proposal

TLDR: Could we add Up option(s) to allow the cli.StartOption.Attach and AttachTo to be set.

> Ref Version v0.31.0

The current implementation of "github.com/testcontainers/testcontainers-go/modules/compose" does not provide a Up  option (StackUpOption) to set either the cli.StartOptions.Attach or AttachTo properties.   [Up func](https://github.com/testcontainers/testcontainers-go/blob/main/modules/compose/compose_api.go#L316-L319) 

My proposal is we add 2 extra options to allow the cli.Attach and cli.AttachTo properties to be set.   Further these should probably follow the semantics used by the cli `docker compose up` in that if a non nil Attach value is provided but not a AttachTo the AttachTo should default to all services.

With these options it would then be possible to follow logs within the test containers started via a compose Up.   Without this feature the work around is either to poll for logs via a `ServiceContainer` and repeated `Log` call (like the log wait strategy does).   

I would welcome feedback and suggestions of any other work arounds.   If this sounds a sensible way forward happy to contribute with a PR.

**Comment by mdelapenya** [comment]
Hi @nehemming thanks for opening this issue and starting a discussion for it. Have you considered using the log consumer concept that we already have for generic containers? If we add that option, we could automatically start a log consumer config for the given service, so that the client code could define its own consumer (in-memory, to a database, etc) and simply consume the logs in the same way that the generic containers do.

Wdyt?

**Comment by nehemming** [comment]
Thanks for the reply, I agree making the config analogous to the generic containers would be good - I suggested an option as that seemed the existing mechanism to customise.   I will take a look at it but unless I am mistaken will ultimately need use the cli startOptions Attach property.   I will do some experimenting and report back.   However if you know or suspect this another way replicate the generic container behaviour you may prevent me going down a dead end.
